### ISSUE TITLE
pcl_catkin && catkin build dcl_fast_lio error

### ISSUE THREAD

**Issue by Lee-JaeWon** [PengYu-Team/DCL-SLAM#18]
I tried running dcl-slam on ROS noetic, but it doesn't work.

I cloned and built [pcl_catkin github](https://github.com/ethz-asl/pcl_catkin) in the src folder because I think I need a pcl-catkin package when I build.

However, when I follow your README step by step, an error occurs in the process of 'catkin build dcl_fast_lio'.

Do you know the solution?

I am confident that I have satisfied both the version and the dependency.

Specifically, I think I need to know what's different between your environment and mine. Especially for the pcl_catkin package.

![Screenshot from 2023-11-01 17-04-34](https://github.com/PengYu-Team/DCL-SLAM/assets/72693388/f292b248-0fb5-46d2-8db7-e56405941495)

To sum up, I only added 'pcl_catkin' to your README method. If I don't add 'pcl_catkin', it says 'pcl_catkin' is required, and if I add it, another error occurs.

This relates to issue #14 .

I'd appreciate it if you could answer.

**Comment by zhongshp** [comment]
Hi~ Lee-JaeWon
Thank you for pointing it out. The current DCL-SLAM code is only compatible with Ubuntu 18.04 Melodic. However, the author also found a compilation problem with the pcl library in Ubuntu 20.04 noetic, which appears to be a conflict between version 1.8 of Pcl-Catkin and ubuntu 20.04 PCL-1.10 (The code for ubuntu 20.04 is coming soon). My suggestion is to skip pcl-catkin and use pcl-1.10, which comes with ubuntu 20.04. Introduction of pcl-1.10 in cmakelists:

find_package(PCL REQUIRED)
include_directories(
${PCL_INCLUDE_DIRS}
)
link_directories(
${PCL_LIBRARY_DIRS}
)

And when compiling the Node executable:
target_link_libraries(Node ${PCL_LIBRARIES})

**Comment by heisetiantang** [comment]
作者您好，很荣幸你能回答之前的问题，关于DCL的复现，还有些问题想向您请教，通过阅读您的论文和代码，一直没有完全弄懂，在三个机器人之间，程序需要传递那几个话题，才能够使得实际机器人，像数据集跑得那样，能够多机器人协同定位建图，或者说通过什么能够证实多个机器人之间成功进行协同，仅仅通过观察地图的变换么，还是在机器人进行回环后，可以接收到其他机器人的位姿和地图，对当前机器人进行补全呢。
![image](https://github.com/PengYu-Team/DCL-SLAM/assets/92718554/a0a55e3d-3061-48de-b523-271d8c33687a)
![image](https://github.com/PengYu-Team/DCL-SLAM/assets/92718554/5aea071e-3d74-4087-a97d-5b8b5d5fb8a0)
![image](https://github.com/PengYu-Team/DCL-SLAM/assets/92718554/2eb9e5ab-bea3-45d8-98ad-84d8ac21fa31)

![image](https://github.com/PengYu-Team/DCL-SLAM/assets/92718554/0afac65a-d1c1-46e8-bbc2-e5bd1e759798)

![image](https://github.com/PengYu-Team/DCL-SLAM/assets/92718554/f8f4c29c-df47-45e9-8283-7d454f8597dc)

**Comment by heisetiantang** [comment]
@zhongshp @Lee-JaeWon

**Comment by Lee-JaeWon** [comment]
@zhongshp
Thank you for your answering.
But ROS melodic also has problems with `pcl_catkin`.
I installed pcl 1.8.1(in ubuntu) and build `fast_lio` by cloning the appropriate `pcl_catkin` in github, but I get an error.
It seems necessary to organize this dependency problem, and I would appreciate it if you could organize it.

LIO-SAM works well. Only fast-lio seems to need a solution.

All were purely produced and executed in **docker** environments.
